docs/prod-maintain-secure-upgrade.md

```diff
@@ -419,21 +419,40 @@ announcement).
   can see future messages sent to the stream, but they do not receive
   access to the stream's message history.
 
-* Zulip supports editing the content or topics of messages that have
-  already been sent (and even updating the topic of messages sent by
-  other users when editing the topic of the overall thread).
-
-  While edited messages are synced immediately to open browser
-  windows, editing messages is not a safe way to redact secret content
-  (e.g. a password) unintentionally shared via Zulip, because other
-  users may have seen and saved the content of the original message
-  (for example, they could have taken a screenshot immediately after
-  you sent the message, or have an API tool recording all messages
-  they receive).
-
-  Zulip stores and sends to clients the content of every historical
-  version of a message, so that future versions of Zulip could support
-  displaying the diffs between previous versions.
+* Zulip supports editing the content and topics of messages that have
+  already been sent. As a general philosophy, our policies provide
+  hard limits on the ways in which message content can be changed or
+  undone. In contrast, our policies around message topics favor
+  usefulness (e.g. for conversational organization) over faithfulness
+  to the original.
+
+  The message editing policy can be configured on the realm
+  administration page. There are three configurations provided out of
+  the box: (i) users cannot edit messages at all, (ii) users can edit
+  any message they have sent, and (iii) users can edit the content of
+  any message they have sent in the last N minutes, and the topic of
+  any message they have sent. In (ii) and (iii), topic edits can also
+  be propogated to other messages with the same original topic, even
+  if those messages were sent by other users. The default setting is
+  (iii), with N = 10.
+
+  In addition, and regardless of the configuration above, some topic
+  edits are always allowed. Messages with no topic can always be
+  edited to have a topic, by anyone in the organization. The topic of
+  any message can also always be edited by a realm administrator.
+
+  Note that the Zulip frontend will sometimes restrict the type of
+  topic edits that can be made, to streamline the user experience. A
+  topic edit not being possible on the frontend does not always mean
+  that restriction is enforced on the backend.
```

Comment:
This paragraph seems unnecessarily detailed compared to the rest of this section, and not that important, so I removed it.